From 49701094fcc8fc510af97e439ad46415342d367f Mon Sep 17 00:00:00 2001
From: Daniel Stenberg <daniel@haxx.se>
Date: Thu, 3 Apr 2025 12:10:45 +0200
Subject: [PATCH] http: fix a build error when all auths are disabled
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

error: ‘result’ undeclared (first use in this function)

Found with randdisable

Closes #16939
---
 lib/http.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

index 03d1959e29ae..51515d1fe628 100644
--- lib/http.c
+++ lib/http.c
@@ -1075,12 +1075,12 @@ CURLcode Curl_http_input_auth(struct Curl_easy *data, bool proxy,
       break;
     Curl_str_passblanks(&auth);
   }
+  return result;
 #else
   (void) proxy;
   /* nothing to do when disabled */
+  return CURLE_OK;
 #endif
-
-  return result;
 }
 
 /**
